tests/IntRangeTest.php

```diff
@@ -457,6 +456,35 @@ function takesInt(int $i): void{
                         return;
                     }',
             ],
+            'intRangeExpandedByLoop' => [
+                'code' => '<?php
+                    for ($i = 0; $i < 10; $i++) {
+                        takesInt($i);
+                    }
+                    for (; $i < 20; $i++) {
+                        takesInt($i);
+                    }
+
+                    /** @psalm-param int<0, 20> $i */
+                    function takesInt(int $i): void{
+                        return;
+                    }',
+            ],
+            'statementsInLoopPreserveNonNegativeIntRange' => [
+                'code' => '<?php
+                    $sum = 0;
+                    foreach ([-6, 0, 2] as $i) {
+                        if ($i > 0) {
+                            $sum += $i;
+                        }
+                    }
+                    takesNonNegativeInt($sum);
+
+                    /** @psalm-param int<0, max> $i */
+                    function takesNonNegativeInt(int $i): void{
+                        return;
+                    }',
+            ],
             'SKIPPED-wrongLoopAssertion' => [
```

Comment:
Can you try to unskip that one too?

Reply:
I'm not sure what that one is trying to test, but it causes the following error:
> Psalm\Exception\CodeException: Trace - src\somefile.php:8:51 - $i: int<0, max>

Reply:
Strange that traces were left in there. Can you get rid of the trace comments to see if there is other errors?

Reply:
There aren't any errors after removing the trace comments.